src/main/scala/SerialAdapter.scala

```diff
@@ -15,18 +15,47 @@ import scala.math.min
 case object SerialAdapter {
   val SERIAL_TSI_WIDTH = 32 // hardcoded in FESVR
 
-  def connectHarnessRAM(serdesser: TLSerdesser, port: ClockedIO[SerialIO], reset: Reset): SerialRAM = {
+  def asyncQueue(port: ClockedIO[SerialIO], clock: Clock, reset: Reset): SerialIO = {
+    val w = port.bits.w
+    // AsyncQueue needs an implicit clock/reset, but they are unused
+    withClockAndReset (false.B.asClock, false.B) {
+      val out_queue = Module(new AsyncQueue(UInt(w.W)))
+      out_queue.io.enq <> port.bits.out
+      out_queue.io.enq_clock := port.clock
+      out_queue.io.enq_reset := reset.asBool
+      out_queue.io.deq_clock := clock
+      out_queue.io.deq_reset := reset.asBool
+      val in_queue = Module(new AsyncQueue(UInt(w.W)))
+      port.bits.in <> in_queue.io.deq
+      in_queue.io.deq_clock := port.clock
+      in_queue.io.deq_reset := reset.asBool
+      in_queue.io.enq_clock := clock
+      in_queue.io.enq_reset := reset.asBool
+      val crossed = Wire(new SerialIO(w))
+      in_queue.io.enq <> crossed.in
+      crossed.out <> out_queue.io.deq
+      crossed
+    }
+  }
+
+  def asyncQueue(port: SerialIO, clock: Clock, reset: Reset): SerialIO = {
```

Comment:
If the inner method is going to be used on the test harness side as an actual asyncqueue  I think we should rename this helper method so people aren't confused when it instantiates a synchronous (but async reset) queue.